### PR TITLE
Initial implementation of the module unification feature flag

### PR DESCRIPTION
**Pull request by mixonic**
Quite barebones and very much a WIP. The mentioned instructions should be helpful for smoke testing:

```
# Install Ember-CLI canary globally:
npm install -g ember-cli/ember-cli
# Create a new app with the module unification blueprint
ember new my-app -b ember-module-unification-blueprint
```

@iezer is following up to push a commit with further tests for the RequireJS ModuleRegistry.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,23 +2,30 @@
...
-    this.options = this.options || {};
-
+  emberResolverFeatureFlags() {
     var config = this.project.config();
     var resolverConfig = config['ember-resolver'] || {};
 
-    var resolverFeatureFlags = Object.assign({
+    return Object.assign({
       /* Add default feature flags here */
+      EMBER_RESOLVER_MODULE_UNIFICATION: true
```
Comment (mixonic) [inline index.js]: This line must be changed before merge. It defaults the FF to `true` when what we want is: `true` only if configured or when running tests.

File: package.json [hunk 2]
```diff
@@ -40,7 +40,11 @@
     "ember-addon"
   ],
   "dependencies": {
+    "@glimmer/resolver": "^0.3.0",
     "babel-plugin-debug-macros": "^0.1.1",
+    "broccoli-babel-transpiler": "^5.6.2",
```
Comment (mixonic) [inline package.json]: I believe this dependency is not required.

Comment (mixonic) [inline package.json]: Additionally `ember-cli` should be version `2.13.0-beta.2`. There is code in `index.js` which is run when the feature flag is enabled (which it should be for tests, but not for tests on older versions of Ember).

File: index.js [hunk 3]
```diff
@@ -2,23 +2,30 @@
...
+
+  init: function() {
+    this._super.init.apply(this, arguments);
+    this.options = this.options || {};
+
+    this._emberResolverFeatureFlags = this.emberResolverFeatureFlags();
 
     this.options.babel = {
       loose: true,
+      blacklist: ['es6.modules'],
```
Comment (rwjblue) [inline index.js]: this is incorrect, babel 6 does not have a blacklist like this

**Pull request merged**